Clarify difference in --check mode

xipmix opened the pull request:
I was going nuts trying to reproduce the example until I realized I was running with --check...

##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->

<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request
 - New Module Pull Request
 - Bugfix Pull Request
 - Docs Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```

```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

xipmix committed:
Clarify difference in --check mode

I was going nuts trying to reproduce the example until I realized I was running with --check...

xipmix committed:
Unbreak the literal block above

Add required whitespace, use the proper notation for notes

bcoca commented on docs/docsite/rst/playbooks_loops.rst:
```diff
@@ -676,6 +676,8 @@ This differs from the data structure returned when using ``register`` without a
         ]
     }
 
+.. note:: The data structure returned when running in ``--check`` mode will look somewhat different.
```
it not always WILL be, but it CAN be, depends on the module. specifically _facts modules should return the same data, but modules that affect change and support check mode, might differ as they didn't really carry out their actions.

xipmix committed:
Generalise the statement about check mode

My original note was in relation to the specific example shown, as that was my experience - running the exact code shown (albeit with ansible 2.2) gave the result shown below - no stdout or stderr fields, no timestamps.

The comment from @bcoca reminded me there will be readers with other mental contexts, ie not focused on the specific example, so I've tried to make the note a bit more general.

```
   {
        "changed": false, 
        "msg": "All items completed", 
        "results": [
            {
                "_ansible_item_result": true, 
                "_ansible_no_log": false, 
                "_ansible_parsed": true, 
                "changed": false, 
                "invocation": {
                    "module_args": {
                        "_raw_params": "echo one", 
                        "_uses_shell": true
                    }, 
                    "module_name": "command"
                }, 
                "item": "one", 
                "msg": "remote module (command) does not support check mode", 
                "skipped": true
            }, 
            {
                "_ansible_item_result": true, 
                "_ansible_no_log": false, 
                "_ansible_parsed": true, 
                "changed": false, 
                "invocation": {
                    "module_args": {
                        "_raw_params": "echo two", 
                        "_uses_shell": true
                    }, 
                    "module_name": "command"
                }, 
                "item": "two", 
                "msg": "remote module (command) does not support check mode", 
                "skipped": true
            }
        ]
    }
```

bcoca commented:
cc @dharmabumstead

dharmabumstead commented:
Feel free to rebase and reopen.
